### PR TITLE
init model

### PR DESCRIPTION
**Pull request by MinhDo2007**

### CODE REVIEW

File: app/models/answer.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Answer < ActiveRecord::Base
+	belongs_to :word
```
Comment (vjnhat) [inline app/models/answer.rb]: indent?

File: app/models/category.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Category < ActiveRecord::Base
+	has_many :lessons
```
Comment (vjnhat) [inline app/models/category.rb]: ident

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class User < ActiveRecord::Base
+  has_many :lessons
+  has_many :active_relationships, class_name:  "Relationship",
+                                  foreign_key: "follower_id",
+                                  dependent:   :destroy
+   has_many :following, through: :active_relationships, source: :followed
```
Comment (khanhkhanh) [inline app/models/user.rb]: indent

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class User < ActiveRecord::Base
+  has_many :lessons
+  has_many :active_relationships, class_name:  "Relationship",
+                                  foreign_key: "follower_id",
+                                  dependent:   :destroy
```
Comment (khanhkhanh) [inline app/models/user.rb]: spaces between dependent: and :destroy

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class User < ActiveRecord::Base
+  has_many :lessons
+  has_many :active_relationships, class_name:  "Relationship",
```
Comment (khanhkhanh) [inline app/models/user.rb]: spaces

File: app/models/user.rb (hunk 4 above)
Comment (phamkykhoi) [inline app/models/user.rb]: > 1 space

File: app/models/category.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+class Category < ActiveRecord::Base
+  has_many :lessons
+  has_many :words
```
Comment (phamkykhoi) [inline app/models/category.rb]: Thiếu

File: app/models/lesson.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class Lesson < ActiveRecord::Base
+  belongs_to :user
```
Comment (vjnhat) [inline app/models/lesson.rb]: refactor code,you should divide it to be 2 component,  belongs_to group and has_many group

File: app/models/category.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+class Category < ActiveRecord::Base
+  has_many :lessons
```
Comment (phamkykhoi) [inline app/models/category.rb]: dependent: :destroy

File: app/models/word.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+class Word < ActiveRecord::Base
+  has_many :lessons
+  has_many :answers
```
Comment (phamkykhoi) [inline app/models/word.rb]: dependent: :destroy

File: app/models/user.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+class User < ActiveRecord::Base
+  has_many :lessons
+  has_many :active_relationships, class_name: "Relationship",
+                                  foreign_key: "follower_id",
+                                  dependent: :destroy
+   has_many :following, through: :active_relationships, source: :followed
```
Comment (NguyenTanDuc) [inline app/models/user.rb]: indent

File: app/models/lesson.rb (hunk 7 above)
Comment (TrucLuu) [inline app/models/lesson.rb]: our designed database has no `group` table

**Comment by phamkykhoi** [comment]
OK

**Pull request merged**